app/jobs/socure_image_retrieval_job.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+class SocureImageRetrievalJob < ApplicationJob
+  queue_as :default
+
+  attr_reader :document_capture_session_uuid
+
+  def perform(
+    document_capture_session_uuid:,
+    reference_id:,
+    image_storage_data:
+  )
+    @document_capture_session_uuid = document_capture_session_uuid
+
+    result = fetch_images(reference_id)
+    if result.is_a?(Idv::IdvImages)
+      result.write_with_data(image_storage_data:)
+    else
+      attempts_api_tracker.idv_image_retrieval_failed(
+        **image_storage_data.values.reduce({}) { |a, b| a.merge(b) },
```

Comment:
I think we could consider skipping logging the encryption keys, I'm not sure how useful it is.